xdsl/ir/core.py

```diff
@@ -1677,6 +1684,13 @@ def walk(
         for op in reversed(self.ops) if reverse else self.ops:
             yield from op.walk(reverse=reverse, region_first=region_first)
 
+    def walk_child_blocks_first(self) -> Iterator[Block]:
+        for op in self.ops:
+            for region in op.regions:
+                for block in region.blocks:
+                    yield block
+                    yield from block.walk_child_blocks_first()
```

Comment:
It feels natural for it to have this name, WDYT? It would be worth testing also, probably in the same pytest
```suggestion
    def walk(self) -> Iterator[Block]:
        for op in self.ops:
            for region in op.regions:
                for block in region.blocks:
                    yield block
                    yield from block.walk()
```